### PR TITLE
Create ability to test secret-agent against all browsers

### PR DESCRIPTION
**Pull request by calebjclark**

### CODE REVIEW

File: analyze/lib/Plugin.ts [hunk 1]
```diff
@@ -65,13 +65,13 @@ export default abstract class Plugin {
...
         const toCheck = checksById[probe.check.id];
         const humanScore = probe.check.generateHumanScore(toCheck, profileCountOverTime);
-        // if (probe.id === 'strg-acth') {
-        //   console.log(probe.check.id, probe.check);
-        //   console.log('-----------')
-        //   const toChecks = checks.filter(x => x.idPrefix === probe.check.idPrefix);
-        //   const consoleChecks = toChecks.length ? toChecks : checks.map(x => ({ idPrefix: x.idPrefix, ...x }));
-        //   console.log(toCheck || consoleChecks);
-        // }
+        if (humanScore < 100 && probe.id === 'auto-aaad') {
```
Comment (blakebyrnes) [inline analyze/lib/Plugin.ts:68]: meant to be hardcoded?

File: external-data/lib/TypeSerializer.ts [hunk 2]
```diff
@@ -0,0 +1,151 @@
+export default class TypeSerializer {
```
Comment (blakebyrnes) [inline external-data/lib/TypeSerializer.ts:1]: I can't quite figure out where this is getting used. Any thoughts on how we'll keep these in sync?

File: runner/scripts/0-collectFoundationalProfiles.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    const userAgent = RealUserAgents.getId(userAgentId);
+    if (!userAgent) throw new Error(`${userAgentId} not supported by RealUserAgents`);
+
+    const fromDir = `${slabProfilesDir}/${userAgentId}`;
+    const toDir = Path.join(__dirname, `../data/external/0-foundational-profiles/${userAgentId}`);
+    copyDir(fromDir, toDir);
+  }
+}
+
+// async function collectLocalProfiles() {
```
Comment (blakebyrnes) [inline runner/scripts/0-collectFoundationalProfiles.ts:29]: Commented out code. Just making sure it's on purpose

File: runner/scripts/2-collectUserAgentsToTest.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import * as Fs from 'fs';
+import * as Path from 'path';
+import RealUserAgents from "@double-agent/real-user-agents";
+import IUserAgentToTest, {UserAgentToTestPickType} from "@double-agent/config/interfaces/IUserAgentToTest";
+// import localUserAgentConfig from '../data/local/userAgentConfig.json';
+import externalUserAgentConfig from '../data/external/userAgentConfig.json';
+
+const dataDir = Path.join(__dirname, '../data');
+
+// const localBaseDir = `${dataDir}/local`;
```
Comment (blakebyrnes) [inline runner/scripts/2-collectUserAgentsToTest.ts:10]: Commented out code. Just making sure it's on purpose

File: runner/scripts/4-analyzeAssignmentResults.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+const userAgentIds = Fs.readdirSync(`${assignmentsDataDir}/individual`);
+const analyze = new Analyze(userAgentIds.length, probesDataDir);
+
+for (const userAgentId of userAgentIds) {
+  const flags = analyze.addIndividual(`${assignmentsDataDir}/individual`, userAgentId);
+  const saveFlagsToDir = Path.resolve(resultsDir, userAgentId);
+  saveFlagsToPluginFiles(saveFlagsToDir, flags);
+}
+
+// for (const pickType of [UserAgentToTestPickType.popular, UserAgentToTestPickType.random]) {
```
Comment (blakebyrnes) [inline runner/scripts/4-analyzeAssignmentResults.ts:22]: Commented out code. Just making sure it's on purpose

File: external-data/lib/TypeSerializer.ts (hunk 2 above)
Comment (calebjclark) [reply external-data/lib/TypeSerializer.ts:1]: It's used in TimeoutError which is used in Resolvable which is used in createPromise which is used in downloadFile.

Comment (calebjclark) [reply external-data/lib/TypeSerializer.ts]: Removed

File: analyze/lib/Plugin.ts (hunk 1 above)
Comment (calebjclark) [reply analyze/lib/Plugin.ts]: Commented out

**Pull request merged**